vendor-extra/ContentPageBundle/tests/Controller/ContentControllerTest.php

```diff
@@ -47,13 +49,24 @@ public function it_returns_the_title(string $id) : void
             )
         );
 
-        $response = $controller($id);
+        $response = $controller(new Request(), $id);
         $response->prepare(new Request());
-        $crawler = new Crawler($response->getContent());
 
         $this->assertSame(200, $response->getStatusCode());
         $this->assertSame('text/html; charset=UTF-8', $response->headers->get('Content-Type'));
-        $this->assertSame("Article ${id}", $crawler->text());
+        $this->assertJsonStringEqualsJsonString(
+            json_encode(
+                [
+                    'template.html.twig',
+                    [
+                        'lang' => 'en',
+                        'dir' => 'ltr',
+                        'title' => "Article {$id}",
+                    ],
+                ]
+            ),
+            $response->getContent()
```

Comment:
`text/html; charset=UTF-8` but it's a JSON string? :thinking:

Reply:
ah, `TwigTestCase` doesn't really render the template because we are in the context of a bundle; confusing, could use a single HTML element in which the JSON is dumped (and would leave space to dump other things)

Reply:
Makes it harder to compare (string matching could be brittle). Agree it looks a bit confusing on first glance.

Reply:
String matching is brittle, but it's possible to match the element and extract the inner text to parse it with `json_decode`; it doesn't look like this is a test that needs to be replicated many times, and even so the access to the right element can be extracted into a private method provided by `TwigTestCase` itself to keep all the knowledge about this in a single place